### DISCUSSION TITLE
ImportError: cannot import name 'BleakScanner'

### DISCUSSION THREAD

**Discussion by Gladiatech** [hbldh/bleak#516] [Q&A]
* bleak version:0.11.0
* Python version:3.8.4
* Operating System:Windows 10 19041.867
* BlueZ version (`bluetoothctl -v`) in case of Linux:

### Description

I've just installed bleak, trying to execute the first script 

```python
import asyncio
from bleak import BleakClient

address = "24:71:89:cc:09:05"
MODEL_NBR_UUID = "00002a24-0000-1000-8000-00805f9b34fb"

async def run(address):
    async with BleakClient(address) as client:
        model_number = await client.read_gatt_char(MODEL_NBR_UUID)
        print("Model Number: {0}".format("".join(map(chr, model_number))))

loop = asyncio.get_event_loop()
loop.run_until_complete(run(address))
```

I get : ImportError: cannot import name "BleakScanner" from partially initialized module 'bleak' (most likely due to a circular import) (C:\Users\Fra\Desktop\bleak.py)

I tried to uninstall/reinstall again and change python version from 3.9 but not luck

cheers

**Comment by dlech** [comment]
Do not name your file  `bleak.py` (or `asyncio.py` or any other module name).

**Reply by Gladiatech** [reply]
Hi,

I tried many names last one trial.py  but no luck 

```
PS C:\Users\Fra> & python c:/Users/Fra/Desktop/trial.py
Traceback (most recent call last):
  File "c:/Users/Fra/Desktop/trial.py", line 2, in <module>
    from bleak import BleakClient
  File "c:\Users\Fra\Desktop\bleak.py", line 2, in <module>
    from bleak import BleakScanner
ImportError: cannot import name 'BleakScanner' from partially initialized module 'bleak' (most likely due to a circular import) (c:\Users\Fra\Desktop\bleak.py)
PS C:\Users\Fra> 
```
```python
import asyncio
from bleak import BleakClient

address = "24:71:89:cc:09:05"
MODEL_NBR_UUID = "00002a24-0000-1000-8000-00805f9b34fb"

async def run(address):
    async with BleakClient(address) as client:
        model_number = await client.read_gatt_char(MODEL_NBR_UUID)
        print("Model Number: {0}".format("".join(map(chr, model_number))))

loop = asyncio.get_event_loop()
loop.run_until_complete(run(address))
```

**Reply by dlech** [reply]
> c:\Users\Fra\Desktop\bleak.py

This says that there is still a file named `bleak.py` that needs to be deleted.

**Reply by Gladiatech** [reply]
Thank you I'm new to Python and I did not realized how it works

**Reply by api2001** [reply]
Wow, I had the same (stupid) issue and the same (stupid having added a bleak.py file just for quick testing) mistake. I am not soo new to Python :thinking:. Thanks for already solving this a few years back in a searchable/findable way :laughing:.
